### PR TITLE
Add guardrails for remote model

### PR DESCRIPTION
**Pull request by jngz-es**
### Description
Update documents with guardrails for remote model.

### Issues Resolved
Closes [6743](https://github.com/opensearch-project/documentation-website/issues/6743)

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kolchfa-aws** [comment]
Thank you, @jngz-es! I have edited the PR; please review my edits. Additionally, I have some questions about the usage of guardrails:
- Does `source_fields` specify the names of the fields in the documents contained in the `index_name` index? 
- What format do `source_fields` have to have? Let's say my field is `test_field`. Should it contain a list of strings?
- Does the user have to upload the documents with the stopwords before registering the model?
- Is it only the predict request that is rejected if guardrails are triggered?
Thanks.

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 1]
```diff
@@ -248,6 +249,68 @@ OpenSearch responds with the `task_id` and task `status`.
...
+  }
+}
+```
+{% include copy-curl.html %}
+
+For an end-to-end example, see [Guardrails]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/guardrails/).
+
+#### Example response
+
+OpenSearch responds with the `task_id` and task `status`.
```
Comment (natebower) [inline _ml-commons-plugin/api/model-apis/register-model.md:305]:
```suggestion
OpenSearch responds with the `task_id` and task `status`:
```

File: _ml-commons-plugin/remote-models/guardrails.md [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+nav_order: 70
+parent: Connecting to externally hosted models 
+grand_parent: Integrating ML models
+---
+
+# Configuring model guardrails
+**Introduced 2.13**
+{: .label .label-purple }
+
+Guardrails can guide a large language model (LLM) toward desired behavior. They act as a filter, preventing the LLM from generating outputs that are harmful or violate ethical principles and facilitating a safer use of AI. Guardrails also steer the LLM to produce a more focused and relevant output.
```
Comment (natebower) [inline _ml-commons-plugin/remote-models/guardrails.md:15]:
```suggestion
Guardrails can guide a large language model (LLM) toward desired behavior. They act as a filter, preventing the LLM from generating output that is harmful or violates ethical principles and facilitating safer use of AI. Guardrails also cause the LLM to produce more focused and relevant output. 
```

File: _ml-commons-plugin/remote-models/guardrails.md [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+
+To configure guardrails for your LLM, you can provide a list of words that are prohibited in the input or output of the model. Alternatively, you can provide a regular expression against which the model input or output will be matched.
+
+## Prerequisites
+
+Before you start, make sure you have fulfilled the [prerequisites]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/index/#prerequisites) for connecting to an externally hosted model.
+
+## Step 1: Create a guardrail index
+
+To start, create an index that will hold the excluded words (_stopwords_). In the index settings, specify for the index to contain a `title` field, which will contain excluded words, and a `query` field, which is of the [percolator]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/percolator/) type. The percolator query will be used to match the LLM input or output:
```
Comment (natebower) [inline _ml-commons-plugin/remote-models/guardrails.md:25]:
```suggestion
To start, create an index that will store the excluded words (_stopwords_). In the index settings, specify a `title` field, which will contain excluded words, and a `query` field, which is of the [percolator]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/percolator/) type. The percolator query will be used to match the LLM input or output:
```

File: _ml-commons-plugin/remote-models/guardrails.md [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+ "model_group_id": "wlcnb4kBJ1eYAeTMHlV6",
+ "status": "CREATED"
+}
+```
+
+To learn more about model groups, see [Model access control]({{site.url}}{{site.baseurl}}/ml-commons-plugin/model-access-control/).
+
+## Step 4: Create a connector
+
+Create a connector to the model. In this example, you'll create a connector to the Anthropic Claude model hosted on Amazon Bedrock:
```
Comment (natebower) [inline _ml-commons-plugin/remote-models/guardrails.md:100]: "Now you can create a connector for the model"?

File: _ml-commons-plugin/remote-models/index.md [hunk 5]
```diff
@@ -317,3 +317,4 @@ To learn how to use the model for vector search, see [Using an ML model for neur
 - For more information about connector parameters, see [Connector blueprints]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/blueprints/).
 - For more information about managing ML models in OpenSearch, see [Using ML models within OpenSearch]({{site.url}}{{site.baseurl}}/ml-commons-plugin/model-serving-framework/).
 - For more information about interacting with ML models in OpenSearch, see [Managing ML models in OpenSearch Dashboards]({{site.url}}{{site.baseurl}}/ml-commons-plugin/ml-dashboard/)
+For steps to configure model guardrails, see [Guardrails]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/guardrails/)
```
Comment (natebower) [inline _ml-commons-plugin/remote-models/index.md:320]:
```suggestion
For instructions on how to configure model guardrails, see [Guardrails]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/guardrails/).
```

**Review by natebower — COMMENTED** [review]
@jngz-es @kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 6]
```diff
@@ -248,6 +249,68 @@ OpenSearch responds with the `task_id` and task `status`.
...
+```json
+POST /_plugins/_ml/models/_register
+{
+  "name": "openAI-gpt-3.5-turbo",
+  "function_name": "remote",
+  "model_group_id": "1jriBYsBq7EKuKzZX131",
+  "description": "test model",
+  "connector_id": "a1eMb4kBJ1eYAeTMAljY",
+  "guardrails": {
+    "input_guardrail": {
```
Comment (dbwiddis) [inline _ml-commons-plugin/api/model-apis/register-model.md:278]: Our testing has shown that a "type" field is required here or it fails parsing.  @jngz-es @ylwu-amzn

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 7]
```diff
@@ -248,6 +249,68 @@ OpenSearch responds with the `task_id` and task `status`.
...
+        }
+      ],
+      "regex": ["regex1", "regex2"]
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+For an end-to-end example, see [Guardrails]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/guardrails/).
```
Comment (kolchfa-aws) [inline _ml-commons-plugin/api/model-apis/register-model.md:301]:
```suggestion
For a complete example, see [Guardrails]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/guardrails/).
```

File: _ml-commons-plugin/remote-models/guardrails.md [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
```
Comment (kolchfa-aws) [inline _ml-commons-plugin/remote-models/guardrails.md:100]:
```suggestion
Now you can create a connector for the model. In this example, you'll create a connector to the Anthropic Claude model hosted on Amazon Bedrock:
```

File: _ml-commons-plugin/api/model-apis/register-model.md (hunk 6 above)
Comment (kolchfa-aws) [reply _ml-commons-plugin/api/model-apis/register-model.md:280]: Thank you, @dbwiddis! I have added the `type` parameter in this commit https://github.com/opensearch-project/documentation-website/pull/6750/commits/2b13532d4d1a957663fd8e3e4037d1fb254fc670

**Review by kolchfa-aws — APPROVED** [review]

**Pull request merged**